components/playlist/playlist_download_request_manager.h

```diff
@@ -0,0 +1,89 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_COMPONENTS_PLAYLIST_PLAYLIST_DOWNLOAD_REQUEST_MANAGER_H_
+#define BRAVE_COMPONENTS_PLAYLIST_PLAYLIST_DOWNLOAD_REQUEST_MANAGER_H_
+
+#include <list>
+#include <memory>
+#include <string>
+
+#include "base/memory/weak_ptr.h"
+#include "base/scoped_observer.h"
+#include "brave/components/playlist/playlist_types.h"
+#include "brave/components/playlist/playlist_youtubedown_component_manager.h"
+#include "content/public/browser/web_contents.h"
+#include "content/public/browser/web_contents_observer.h"
+
+namespace base {
+class Value;
+}  // namespace base
+
+namespace content {
+class BrowserContext;
+}  // namespace content
+
+namespace playlist {
+
+struct CreatePlaylistParams;
+
+class PlaylistDownloadRequestManager :
```

Comment:
mb this is PlaylistDownloadHelper?

Reply:
I think it's not TabHelper. This only observes one WebContents created for injecting youtubedown.js script.